### ISSUE TITLE
Documentation

### ISSUE THREAD

**Issue by nicholas-leonard** [torch/optim#65]
Each function should be documented in the README, instead of just inline. There should also be a link for each optimization function to the original paper.

**Comment by scabi** [comment]
Made some small progress on documentation. Let me know what you think and what is missing.

**Comment by nicholas-leonard** [comment]
@scabi That looks awesome! Wouldn't call it small.

**Comment by scabi** [comment]
Thanks! It was mostly copy paste from the code though. Any more ideas to improve the documentation? I am willing to help in the next few months.

**Comment by nicholas-leonard** [comment]
@scabi The next thing would be more examples. You could add an examples/ folder and put some concrete examples in there.

**Comment by scabi** [comment]
I think some iTorch notebooks would be the best option for examples. Torch roadmap document says "Tutorials: provide guidelines and basic framework/standard to write and publish tutorials?" Is there any progress on this?

**Comment by nicholas-leonard** [comment]
Nope. Which means you can pretty much do it as you want, subject to the constraint that your PR gets merged XD

**Comment by scabi** [comment]
Got it. I'll look into it.
